Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Revenue.hs

```diff
@@ -61,13 +64,18 @@ getAllDriverFeeHistory merchantShortId mbFrom mbTo = do
     getNumRides fee = sum $ map numRides fee
     getTotalAmount fee = sum $ map (\fee_ -> fromIntegral fee_.govtCharges + fee_.platformFee.fee + fee_.platformFee.cgst + fee_.platformFee.sgst) fee
 
-getCollectionHistory :: ShortId DM.Merchant -> Maybe Text -> Maybe Text -> Maybe UTCTime -> Maybe UTCTime -> Flow Common.CollectionList
-getCollectionHistory merchantShortId volunteerId place mbFrom mbTo = do
+getCollectionHistory :: ShortId DM.Merchant -> Maybe Text -> Maybe Text -> Maybe UTCTime -> Maybe UTCTime -> Maybe Common.Basis -> Flow Common.CollectionList
+getCollectionHistory merchantShortId volunteerId place mbFrom mbTo basis = do
   now <- getCurrentTime
   merchant <- findMerchantByShortId merchantShortId
   let defaultFrom = UTCTime (fromGregorian 2020 1 1) 0
       from_ = fromMaybe defaultFrom mbFrom
       to = fromMaybe now mbTo
+  let rangeBasis = case basis of
+        Just Common.Daily -> KS.Days
+        Just Common.Hourly -> KS.Hours
+        Nothing -> KS.Days
+  when (rangeBasis == KS.Hours && diffUTCTime to from_ > 24 * 60 * 60) $ throwError $ InvalidRequest "Hourly basis data can be seen only in the range of one day "
```

Comment:
I think we can remove the query param and just show on hourly basis if its a day and daily basis if more than one day. Can you make that change? I'll merge the PR after that